[CMS-541, CMS-547] Add announcement banner definition to HeroBanner, do follow up updates, Investigate using link reference for Hero component CTA

levadadenys opened the pull request:
<table><tr><td>
<h2> Warning!! </h2>

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2025. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.
</td></tr></table>

<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->

### [CMS-541, CMS-547] Add announcement banner definition to HeroBanner, do follow up updates, Investigate using link reference for Hero component CTA

* [feat(heroBanner): add announcementBanner support, minor updates](https://github.com/code-dot-org/code-dot-org/commit/86061b278d673de3ed97f22b10be18ef8b325ac0)
* [feat(heroBanner): make buttonLink single prop instead of having multiple props for link](https://github.com/code-dot-org/code-dot-org/commit/4d2b7514747a9fbf9a019fd43c74e5d75068703b)
* [feat(heroBanner): reorder props to match figma](https://github.com/code-dot-org/code-dot-org/commit/08d8bfcfa71874090a0505450f34be7531c50357)


https://github.com/user-attachments/assets/ef8983cc-d4ea-4bfc-8b77-87e367c9d247



## Links

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

- Jira: [CMS-541](https://codedotorg.atlassian.net/browse/CMS-541), [CMS-547](https://codedotorg.atlassian.net/browse/CMS-547)

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

kelbyhawn commented:
> Only tiny nit is I'm noticing that the link in the announcement banner is slightly misaligned with text

This should be able to  be fixed by changing [this](https://github.com/code-dot-org/code-dot-org/blob/aa0dae58025f375aa1affe08fc2b5ed88e100ec6/frontend/packages/component-library/src/alert/alert.module.scss#L16) to `align-items: baseline`

kelbyhawn reviewed (commented):
Looks good overall, but have a couple fixes re: the button in the Hero Banner part:
- The spacing between the button and the paragraph in the Hero Banner seems too like too much, I'll let @moshebaricdo speak to this, but I think it'd look better with `margin: 1rem 0 0`
- It's showing the external link icon when the link in the content entry is not external 

| Before spacing | After spacing |
| ---- | ---- |
| <img width="859" alt="Before" src="https://github.com/user-attachments/assets/08487804-6d8e-41a7-bd59-b4f8b9ad1f1e" /> | <img width="859" alt="After" src="https://github.com/user-attachments/assets/80501dac-a2be-4fd4-8d70-075b562ddaf8" /> |

levadadenys commented:
> The spacing between the button and the paragraph in the Hero Banner seems too like too much, I'll let @moshebaricdo speak to this, but I think it'd look better with margin: 1rem 0 0

That's a know issuee we adress in a [separate ticket/pr](https://codedotorg.atlassian.net/browse/CMS-545)


> It's showing the external link icon when the link in the content entry is not external

Thank you for catching this! There was a mistake in code, updated and fixed this issue
